registration/views.py

```diff
@@ -9,14 +9,72 @@
 from django.conf import settings
 from django.utils.decorators import method_decorator
 from django.utils.module_loading import import_string
+from django.utils.http import is_safe_url
 from django.views.decorators.debug import sensitive_post_parameters
 
 REGISTRATION_FORM_PATH = getattr(settings, 'REGISTRATION_FORM',
                                  'registration.forms.RegistrationForm')
 REGISTRATION_FORM = import_string(REGISTRATION_FORM_PATH)
 
 
-class RegistrationView(FormView):
+class _RequestPassingFormView(FormView):
+    """
+    A version of FormView which passes extra arguments to certain
+    methods, notably passing the HTTP request nearly everywhere, to
+    enable finer-grained processing.
+
+    """
+    def _clean_next_param(self, next=None):
+        # removes xss attacks from next param
+        # unfortunatly can't just remove it from
+        if next and ('"' in next or '\'' in next or not is_safe_url(next)):
+            raise ValueError('Invalid request param')
+        return next
+
+    def get(self, request, *args, **kwargs):
+        # Pass request to get_form_class and get_form for per-request
+        # form control.
+        next = self._clean_next_param(request.GET.get('next'))
+        form_class = self.get_form_class(request)
+        form = self.get_form(form_class)
+        context = self.get_context_data(form=form)
+        if next:
+            context.update(dict(next=next))
+        return self.render_to_response(context)
+
+    def post(self, request, *args, **kwargs):
```

Comment:
you shouldn't ahve to redefine all of the subclass methods if you are just calling `super` on them essentially.